internal/blueprint/repository_customizations.go

```diff
@@ -324,3 +327,73 @@ func validateCustomRepository(repo *RepositoryCustomization) error {
 	}
 	return nil
 }
+
+func (rc *RepositoryCustomization) getFilename() string {
+	if rc.Filename == "" {
+		return fmt.Sprintf("%s.repo", rc.Id)
+	}
+	return rc.Filename
+}
+
+func RepoCustomizationsToRepoConfigAndGPGKeyFiles(repos []RepositoryCustomization) (map[string][]rpmmd.RepoConfig, []*fsnode.File, error) {
+	if len(repos) == 0 {
+		return nil, nil, nil
+	}
+
+	repoMap := make(map[string][]rpmmd.RepoConfig, len(repos))
+	gpgKeyFiles := []*fsnode.File{}
```

Comment:
IMO it would be better to declare the variable without assignment (so `var gpgKeyFiles []*fsnode.File`, so that if there are no Files for GPG keys after the conversion, the returned value will be `nil`. Checking the `len()` of such value will work fine, but you'll need to adjust unit tests.